runtime/lua/vim/ui.lua

```diff
@@ -0,0 +1,32 @@
+local M = {}
+
+--- Prompts the user to pick a single item from a collection of entries
+---
+---@param entries table Arbitrary entries
+---@param opts table Additional options
```

Comment:
```suggestion
---@param opts table Options described at |input()|, plus:
```

should probably map to the `{opts}` supported by input() , to start with?

Reply:
It's using `inputlist` under the hood, not `input`. And I'd consider `inputlist` an implementation detail.

And I'd rather start small and add more options as use-cases arise instead of anticipating more up-front.

Considering that this is intended as an extension point to be overridden we also have to consider what's easy or hard to implement. E.g. if you want to dispatch to a tool like `fzf`.

Reply:
> And I'd rather start small and add more options as use-cases arise instead of anticipating more up-front.

that's fine, the point is to mark the missing options as TODO so that we don't invent similar but different options in the future. Use input() as a reference point unless there is a conscious reason to diverge.

> Considering that this is intended as an extension point to be overridden we also have to consider what's easy or hard to implement. E.g. if you want to dispatch to a tool like fzf.

- Is there also a goal of having an easy-to-use interface? If so, those are in conflict.
-  if we have a sparse interface that is easy to override, does that risk it being ignored in favor of full-featured implementations?
- can "easy to override" be achieved by allowing partial implementations ?

Reply:
> Sure, but that's orthogonal to asking if there is a reason to diverge from its interface as a basic model.

But they are are different functions.

Looking at the `opts` of `input`:

`default` as string value doesn't make sense. What would make sense is to indicate which item should be preselected. But I think `pick_one` (or whatever we call it) should also work as fuzzy filter - given that one of the use-cases is to allow telescope to override it, and many people use telescope as generic fuzzy finder picker.

`completion` - doesn't make sense in the context of choosing one item from many(?)

`cancelreturn` - might make sense, though I think it is kinda redundant as a feature and the word `return` is a poor fit given that the `pick_one` function uses a callback.

`highlight` - May be an option for the future. I don't know.



> Sure, the point is to mark the missing options as TODO so that we don't invent similar but different options in the future.

I can add a TODO in the function body, but I wouldn't put it in the user-facing docs?


> Is there also a goal of having an easy-to-use interface? If so, those are in conflict.

It is a goal and I don't see how it is in conflict. It adds some constraints in how powerful the function can get.

> if we have a sparse interface that is easy to override, does that risk it being ignored in favor of full-featured implementations?

Possibly to some degree. I think input from @tjdevries  would be very valueable here due to the experience with Telescope. 


> can "easy to override" be achieved by allowing partial implementations ?

Possibly, but then we've to make sure that they degrade to something that is still working in a basic form.